include/srexception.h

```diff
@@ -223,6 +254,14 @@ class DatabaseException: public Exception
     SRError to_error_code() const noexcept override {
         return SRDatabaseError;
     }
+
+    /*!
+    *   \brief Get a string representation of the exception class
+    *   \returns Stringified version of the class name
+    */
+    virtual std::string exception_class() {
+        return std::string("DatabaseException")
```

Comment:
I had to google this, but would something like [this](https://stackoverflow.com/questions/3649278/how-can-i-get-the-class-name-from-a-c-object) avoid refactor/rename/copypasta errors from reporting incorrect class names?

```
return typeid(theObject).name();
```

Reply:
GCC gives back a mangled name for this, which will not be very user friendly, so this isn't going to be viable.